internal/website/content/howto/docstore/actions.md

```diff
@@ -0,0 +1,108 @@
+
+<!--more-->
+
+## Representing Documents
+
+We'll use a collection with documents represented by this Go struct:
+
+```go
+type Player struct {
+	Name             string
+	Score            int
+	DocstoreRevision interface{}
+}
+```
+
+We recommend using structs for documents because they impose some structure on
+your data, but Docstore also accepts `map[string]interface{}` values. See
+[the `docstore` package
+documentation](https://godoc.org/gocloud.dev/internal/docstore#hdr-Documents)
+for more information.
+
+The `DocstoreRevision` field holds information about the latest revision of the
+document. We discuss it [below]({{< ref "#docrev" >}}).
+
+## Actions and Action Lists
+
+Once you have [opened a collection]({{< ref "open-collection.md" >}}), you
+can call action methods on it. We will use `coll` as the variable holding the collection.
+
+Docstore supports six actions on documents:
+
+- `Get` retrieves a document.
+- `Create` creates a new document.
+- `Replace` replaces an existing document.
+- `Put` puts a document whether or not it is already present.
+- `Update` applies a set of modifications to a document.
+- `Delete` deletes a document.
+
+You can create a single document with the `Collection.Create` method:
+
+```go
+err := coll.Create(ctx, &Player{Name: "Pat", Score: 10})
+if err != nil {
+    return err
+}
+```
+
+If you have more than one action to perform, it is better to use an action
+list. Drivers can optimize action lists by using bulk RPCs, running the actions
+concurrently, or employing a provider's special features to improve efficiency
+and reduce cost. Here we create several documents using an action list.
+
+{{< goexample "gocloud.dev/internal/docstore.ExampleCollection_Actions_bulkWrite" >}}
+
+`ActionList` has a fluent API, so you can build and execute a sequence of
+actions in one line of code. Here we `Put` a document and immediately `Get` its new
+contents.
+
+{{< goexample "gocloud.dev/internal/docstore.ExampleCollection_Actions_getAfterWrite" >}}
+
+(If the underlying provider is eventually consistent, the result of the `Get`
+might not reflect the `Put`. Docstore only guarantees that it will perform the
+`Get` after the `Put` completes.)
+
+See the documentation for [`docstore.ActionList`][] for the semantics of action
+list execution.
+
+[`docstore.ActionList`]: https://godoc.org/gocloud.dev/internal/docstore#ActionList
+
+## Updates
+
+Use `Update` to modify individual fields of a document. The `Update` action takes a
+set of modifications to document fields, and applies them all atomically. You
+can change the value of a field, increment it, or delete it.
+
+{{< goexample "gocloud.dev/internal/docstore.ExampleCollection_Update" >}}
+
+## Revisions {#docrev}
+
+Docstore maintains a revision for every document. Whenever the document is
+changed, the revision is too. By default, Docstore stores the revision
+in a field named `DocstoreRevision`.
```

Comment:
maybe add something about how to change the default? Even if it's just something like "; the `Collection` constructors generally take an `Option` to use a different field."

Reply:
done